content/posts/2021/reana-client-0.7.3.md

```diff
@@ -0,0 +1,346 @@
+verify if the image exists locally and remotely. In this particular
+case, since we do not have the `python:latest` image pulled locally,
+a warning is displayed. On the other hand, the `python:latest` image
+exists in Docker Hub registry, this we get a success message.
+
+
+The last warning displayed is related to the user ID (UID) and
+group IDs (GID) validation. The workflows executed in the REANA
+cluster use containerised environment images to run the steps. Each
+image needs a [user ID and group IDs](https://docs.docker.com/engine/reference/builder/#user)
+to execute the instructions. By default, REANA uses GID 0 (root
+group) and UID 1000, so the [image user must be added to the root group](https://docs.reana.io/advanced-usage/containers/docker/#supporting-arbitrary-user-ids).
+This check attempts to ensure that this guideline is followed,
+otherwise, it may cause problems with permissions when running the
+workflow.
+
+This check is only possible if the environment image is available
+locally, this is why it suggests using the `--pull` option to pull
+the image and be able to proceed. Note that this operation might take
+some minutes, depending on the image size and your network connection.
+
+Let us now use a tagged image and pass the `--pull` option to be able
+to run the UID/GIDs verifications:
+
+```diff
+   specification:
+     steps:
+       - name: run-script
+-        environment: 'python:latest'
++        environment: 'python:3.9-slim'
+         commands:
+           - python "${script_path}" "${sleeptime}"
+```
+
+```console
+$ reana-client validate -f reana.yaml --environments --pull
+==> Verifying REANA specification file... my-analysis/reana.yaml
+  -> SUCCESS: Valid REANA specification file.
+==> Verifying environments in REANA specification file...
+  -> SUCCESS: Environment image python:3.9-slim has the correct format.
+  -> WARNING: Environment image python:3.9-slim does not exist locally.
+  -> SUCCESS: Environment image python:3.9-slim exists in Docker Hub.
+Unable to find image 'python:3.9-slim' locally
+3.9-slim: Pulling from library/python
+6f28985ad184: Pulling fs layer
+...
+Status: Downloaded newer image for python:3.9-slim
+  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
+```
+
+Now we can see that the image is pulled locally and we have a
+different output for the UID/GIDs check. In this case, it is just
+warning us that REANA uses UID 1000 by default, and UID 0 was found.
+This does not necessarily mean that the execution is going to fail,
+but that it would be ideal to run the image using UID 1000 instead.
+In the end, what is more important about this check is the fact that
+the user belongs to the root GID (0). In that case, we would have
+got an error and the validation would have failed.
+
+The workflow environment images validation is also compatible with
+images hosted in the GitLab registry at CERN.
```

Comment:
Perhaps give an example in parentheses, 

... in the GitLab registry at CERN, for example `gitlab-registry.cern.ch/johndoe/foo`. 

And also and with a phrase alluding to privacy, like:

Please note that if your image is protected, you would have to `docker login ...` first so that the validator would be able to fetch it on the machine where you are executing the validation.